### PR TITLE
Clean up bench-tps client code and move to own crate

### LINKED ISSUE #1488: Bench-TPS client is messy
https://github.com/solana-labs/solana/blob/master/src/bin/bench-tps.rs
Is getting really big and gnarly.  It should be possible to break it up into a library with well tested primitives and a binary.

### PR DESCRIPTION
**Pull request by MarkJr94**
#### Problem
Bench-TPS client code is messy and un-tested

#### Summary of Changes
Separate CLI/clap related code, create a new `Config` struct to hold all
configuration/CLI args

Remove most code from `main.rs`

Add a little documentation

Fixes #1488

### CODE REVIEW

**Comment by MarkJr94** [comment]
I would like to write more documentation and/or tests for this tool but I would need help for that most likely

**Comment by mvines** [comment]
This is great!  There's a couple changes that we need to land this PR:
* [ ] CI failed because the `solana-bench-tps` executable is now called `bench_tps`.   Let's just set `name = "solana-bench-tps"` in `bench_tps/Cargo.toml` to keep the rest of the tree happy
* [ ] `vote-signer/` and `multinode-demo/` set a precedent of kebab-case, so let's de-snake `bench_tps/` info `bench-tps/'

There's a couple other random places that we need to touch in the tree when adding a new crate:
* [ ] Please add bench-tps into the list of `CRATES at ci/publish-crate.sh
* [ ] Same for `scripts/cargo-install-all.sh`

**Comment by MarkJr94** [comment]
@mvines I've done everything except making changes to `scripts/cargo-install-all.sh`. It does not exist yet on my branch. I was going to rebase and make the change but does this still make sense in light of what's in `cargo-install-all.sh` as of commit 088bab61a47ac34a67ee111e5aacf2565a27ab34 ? ([Link here](https://github.com/solana-labs/solana/blob/088bab61a47ac34a67ee111e5aacf2565a27ab34/scripts/cargo-install-all.sh))

File: ci/publish-crate.sh [hunk 1]
```diff
@@ -36,12 +36,17 @@ if [[ -n $BUILDKITE_TAG && -n $TRIGGERED_BUILDKITE_TAG ]]; then
...
+<<<<<<< HEAD
 for crate in "${CRATES[@]}"; do
   if [[ ! -r $crate/Cargo.toml ]]; then
     echo "Error: $crate/Cargo.toml does not exist"
     exit 1
   fi
   echo "-- $crate"
+=======
+# shellcheck disable=2044 # Disable 'For loops over find output are fragile...'
+for Cargo_toml in {sdk,metrics,drone,bench-tps,programs/native/{budget,bpf_loader,lua_loader,native_loader,noop,system,vote},.}/Cargo.toml; do
```
Comment (mvines) [inline ci/publish-crate.sh]: Please just add `bench-tps` to the `CRATES` array above.  That's hopefully more natural than this gnary for loop that grew much larger than originally anticipated

**Comment by mvines** [comment]
Yep, change looks good to me.  You should be able to add a `cargo install --path bench-tps "$@"` line after the drone line in `scripts/cargo-install-all.sh` and hopefully be done

**Comment by mvines** [comment]
Hey @MarkJr94, I just realized that we don't really need to publish the new crate so I think it would be fine to just revert the change to  `ci/publish-crate.sh` (which has a merge conflict anyway)

**Comment by garious** [comment]
Can you rebase?

**Comment by mvines** [comment]
Since I keep moving the ground around from under this PR, I decided it would only be fair if I did the final rebase for this effort, https://github.com/solana-labs/solana/pull/2161. Thanks so much for your contribution @MarkJr94!

**Pull request closed without merging**